chalice/awsclient.py

```diff
@@ -92,8 +119,12 @@ def create_function(self,
             kwargs['Timeout'] = timeout
         if memory_size is not None:
             kwargs['MemorySize'] = memory_size
-        return self._call_client_method_with_retries(
-            self._client('lambda').create_function, kwargs)['FunctionArn']
+        try:
+            return self._call_client_method_with_retries(
+                self._client('lambda').create_function, kwargs)['FunctionArn']
+        except Exception as e:
```

Comment:
I still don't think this is right.  We're catching all exceptions and casting them down into a more specific exception (`LambdaClientError`).  If there is actually a bug or something exceptional that happens in the code I'd want the original stack trace to propagate up, not be masked by `LambdaClientError`.